src/Admin/AbstractAdmin.php

```diff
@@ -1365,6 +1362,12 @@ public function generateMenuUrl($name, array $parameters = [], $referenceType =
      */
     public function setTemplates(array $templates)
     {
+        @trigger_error(sprintf(
+            'Method "%s" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0. '
+            .'Use TemplateRegistry services instead.',
```

Comment:
```suggestion
            'Method "%s()" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0.'
            .' Use TemplateRegistry services instead.',
```
BTW, I think the message "Use TemplateRegistry services instead" might be not enough to make clear what the user should do. Is there something more concrete instruction we could suggest? (like call some method, pass that argument or similar).

Reply:
`Use "getTemplateRegistry()->setTemplate()"` will be better.